pkg/agent/util/net.go

```diff
@@ -39,9 +40,9 @@ func generateInterfaceName(key string, name string, useHead bool) string {
 	prefix := name
 	if len(name) > interfacePrefixLength {
 		if useHead {
-			prefix = name[:interfacePrefixLength]
+			prefix = strings.TrimLeft(name[:interfacePrefixLength], "-")
 		} else {
-			prefix = name[len(name)-interfacePrefixLength:]
+			prefix = strings.TrimLeft(name[len(name)-interfacePrefixLength:], "-")
```

Comment:
is there any other character that could be there and would be invalid (e.g. `_`)? Maybe we could have a more generic solution

Reply:
I tried '_', it works fine, but it's actually invalid hostname character, so I suppose we won't see other invalid char since we are using node name and pod name to generate the interface name. a valid hostname is like below:
```
Valid characters for hostnames are ASCII(7) letters from a to z, the digits from 0 to 9, and the hyphen (-). A hostname may not start with a hyphen.
```

Reply:
i think adding this as a comment would help. ie the valid chars